es5-shim.js

```diff
@@ -328,6 +329,70 @@ var strSplit = call.bind(StringPrototype.split);
 var strIndexOf = call.bind(StringPrototype.indexOf);
 var push = call.bind(array_push);
 
+// Tests for inconsistent or buggy `[[Class]]` strings.
+var hasToStringTagBasicBug = toStr() !== '[object Undefined]' || toStr(null) !== '[object Null]';
+var hasToStringTagLegacyArguments = toStr(arguments) !== '[object Arguments]';
+var hasToStringTagInconsistency = hasToStringTagBasicBug || hasToStringTagLegacyArguments;
+// Others that could be fixed:
+// Older ES3 native functions like `alert` return `[object Object]`.
+// Inconsistent `[[Class]]` strings for `window` or `global`.
+
+if (hasToStringTagLegacyArguments) {
+    // To prevent recursion if `Function#call` or `Function#apply` are patched.
```

Comment:
This actually seems to have some strange effect on newer environments (when patch is forced) when not used in conjunction with my `apply` patch. I'm thinking that I may remove it from this this PR.